ocdata/combined.py

```diff
@@ -444,15 +444,15 @@ def generate_random_sites(self, surface_atoms_pos, num_sites):
         """
         dt = scipy.spatial.Delaunay(surface_atoms_pos[:, :2])
         simplices = dt.simplices
-        equal_distr = int(num_sites / len(simplices))
+        equal_distr = max(1, int(num_sites / len(simplices)))
         all_sites = []
         for tri in simplices:
             triangle_positions = surface_atoms_pos[tri]
             sites = self._random_sites_on_triangle(
                 triangle_positions, equal_distr, self.added_z
             )
             all_sites += sites
-        return all_sites
+        return all_sites[:num_sites]
```

Comment:
I assume the ordering of simplices is not random, so if we wanted this to be truly random we could shuffle it